src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  getInitialCards() {
+    //Gets the initial cards
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then((res) => res.json())
+      .then((result) => {
+        return result;
+      })
+      .catch((err) => console.log(err));
+  }
+
+  getCurrentUser() {
+    //Gets the current user's info
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => res.json())
+      .then((result) => {
+        return result;
+      })
+      .catch((err) => console.log(err));
+  }
+
+  updateUser(name, about) {
+    //Updates user information
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    });
+  }
+
+  updateAvatar(avatarUrl) {
+    //Updates the current user's avatar
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar: avatarUrl,
+      }),
+    });
+  }
+
+  createCard(name, link) {
+    //Adding a new card
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        link: link,
+      }),
+    });
```

Comment:
You need to check responses only in `Api` like this: ` .then(this._checkResponse)`. It's responsibility of this class